include/psa/crypto_extra.h

```diff
@@ -1901,6 +1901,331 @@ static inline struct psa_pake_operation_s psa_pake_operation_init( void )
     return( v );
 }
 
+/**
+ * \brief The context for PSA interruptable signing.
+ *
+ * \note  Contents not yet designed as implementation specific.
+ *
+ */
+struct psa_sign_interruptible_context_s
+{
+    unsigned num_ops;
+};
+
+typedef struct psa_sign_interruptible_context_s psa_sign_interruptible_context;
+
+/**
+ * \brief                       Set the maximum number of basic operations done
+ *                              in a row.
+ *
+ *                              If more operations are needed to complete a
+ *                              computation, #PSA_ERROR_IN_PROGRESS will be
+ *                              returned by the function performing the
+ *                              computation. It is then the caller's
+ *                              responsibility to either call again with the
+ *                              same parameters until it returns 0 or an error
+ *                              code; or to free the relevant context if the
+ *                              operation is to be aborted.
+ *
+ *                              It is strictly required that all input
+ *                              parameters and the restart context be the same
+ *                              on successive calls for the same operation,
+ *                              but output parameters need not be the same;
+ *                              they must not be used until the function
+ *                              finally returns 0.
+ *
+ *                              This only applies to functions whose
+ *                              documentation mentions they may return
+ *                              #PSA_ERROR_IN_PROGRESS.
+ *
+ * \param max_ops               Maximum number of basic operations done in
+ *                              a row. Lower (non-zero) values mean signature
+ *                              functions will block for a lesser maximum
+ *                              amount of time. The effect of changing this
+ *                              value will depend on the key used, the curve
+ *                              used and the underlying implementation, so we
+ *                              provide /c
+ *                              psa_sign_interruptible_get_num_ops() in order
+ *                              to help with tuning this value.
+ *
+ */
+void psa_sign_interruptible_set_max_ops( unsigned max_ops );
```

Comment:
Worth adding a comment to say whether we require the user to call this before doing an interruptible operation (I guess we don't and it has a sensible default?)

Reply:
I was thinking there could not be a default as we do not know the underlying implementation, however that would indeed make calling this first a requirement, which I am not a fan of.

I have changed this to be more like the internal implementation, i.e. zero means do the whole job at once, and this value defaults to zero.

Reply:
I don't think that's the right default - by default, users will expect these operations to proceed in small chunks.

Could we dodge this by saying that it is not necessary to call set_max_ops, and the default setting is implementation-defined?

Alternatively, the default could be 1.

Reply:
I have erred on the side of caution, as I think making the implentation initialise this value would be tricky. Thus gone for 1, i.e. the minimum amount of work will be done each call.